pkg/cel/ast/inspector.go

```diff
@@ -256,7 +256,7 @@ func (a *Inspector) inspectCall(call *exprpb.Expr_Call, currentPath string) Expr
 		inspection.FunctionCalls = append(inspection.FunctionCalls, FunctionCall{
 			Name: fmt.Sprintf("%s.%s", a.exprToString(call.Target), call.Function),
 		})
-	} else if !isInternalFunction(call.Function) {
+	} else if !a.env.HasFunction(call.Function) {
```

Comment:
This should imo not be merged unless we have support for optionals as it can lead to side effects for functions which are not properly interpreted with kros cel support. Example is in https://github.com/kro-run/kro/pull/525

Reply:
Are you concerned about backward compatibility. FYI, we are planning to add optionals support :)

Reply:
I was concerned that now Optional functions are allowed but Optional Support is not yet present. I think its fine if we add optionals support as discussed :)